ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/transactions/TransactionsMessageSender.java

```diff
@@ -12,36 +12,36 @@
  */
 package tech.pegasys.pantheon.ethereum.eth.transactions;
 
-import static java.util.stream.Collectors.toSet;
-
 import tech.pegasys.pantheon.ethereum.core.Transaction;
 import tech.pegasys.pantheon.ethereum.eth.manager.EthPeer;
-import tech.pegasys.pantheon.ethereum.eth.messages.TransactionsMessage;
+import tech.pegasys.pantheon.ethereum.eth.messages.LimitedTransactionsMessages;
 import tech.pegasys.pantheon.ethereum.p2p.api.PeerConnection.PeerNotConnected;
 
 import java.util.Set;
+import java.util.stream.StreamSupport;
 
 class TransactionsMessageSender {
 
-  private static final int MAX_BATCH_SIZE = 10;
   private final PeerTransactionTracker transactionTracker;
 
   public TransactionsMessageSender(final PeerTransactionTracker transactionTracker) {
     this.transactionTracker = transactionTracker;
   }
 
   public void sendTransactionsToPeers() {
-    transactionTracker.getEthPeersWithUnsentTransactions().forEach(this::sendTransactionsToPeer);
+    StreamSupport.stream(transactionTracker.getEthPeersWithUnsentTransactions().spliterator(), true)
+        .parallel()
+        .forEach(this::sendTransactionsToPeer);
   }
 
   private void sendTransactionsToPeer(final EthPeer peer) {
     final Set<Transaction> allTxToSend = transactionTracker.claimTransactionsToSendToPeer(peer);
     while (!allTxToSend.isEmpty()) {
-      final Set<Transaction> subsetToSend =
-          allTxToSend.stream().limit(MAX_BATCH_SIZE).collect(toSet());
-      allTxToSend.removeAll(subsetToSend);
+      final LimitedTransactionsMessages limitedTransactionsMessages =
+          LimitedTransactionsMessages.createLimited(allTxToSend);
+      allTxToSend.removeAll(limitedTransactionsMessages.getIncludedTransactions());
```

Comment:
(optional) You could also have LimitedTransactionsMessages.createLimited return a stream of `TransactionMessages`, then you don't need the extra `removeAll` bookkeeping and the extra memory for the included tx's list.